### PR TITLE
bep: templating in parameters schema

### PR DESCRIPTION
**Pull request by benjdlambert**
initial draft for templating in parameters schema

### CODE REVIEW

**Comment by Alok650** [comment]
Hi @benjdlambert do we have a timeline on when this would be released? I have a use-case where I want to use the value of one the parameters as a filter(based on relation) for another parameter.

**Comment by mbenson** [comment]
Not stale you silly bot

**Comment by BillKlineVT** [comment]
Any update on this?

**Comment by benjdlambert** [comment]
@BillKlineVT I just updated a little bit here, hoping to get this reviewed and merged soon :tada:

**Comment by jrjr-dr** [comment]
Will also add that I'm very interested in this functionality. Hope to see it applied soon to Backstage!

**Comment by nia-potato** [comment]
Thanks for doing this @benjdlambert ! hope to see this in backstage soon!

File: beps/0010-scaffolder-templating-in-parameters/README.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+The proposal is to be able to decorate the template schema server side with a context and use that to drive the form rendering client side.
+
+We can extend the `/parameter-schema` endpoint to accept a `formData` context query parameter which will be a JSON object of the current `formData` state.
+
+```diff
+export interface ScaffolderApi {
+  getTemplateParameterSchema(
+    templateRef: string,
++    formData?: string,
```
Comment (freben) [inline beps/0010-scaffolder-templating-in-parameters/README.md:113]: Not really a string at this level right? `JsonObject`?

File: beps/0010-scaffolder-templating-in-parameters/README.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  ): Promise<TemplateParameterSchema>;
+}
+```
+
+```diff
+ router
+    .get(
```
Comment (freben) [inline beps/0010-scaffolder-templating-in-parameters/README.md:120]: might want to support POST here if the formData can become large (eg when having file inputs or entity pickers that match a large part of the catalog or whatnot)

File: beps/0010-scaffolder-templating-in-parameters/README.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+You can see a quick implementation of this in this [branch](https://github.com/backstage/backstage/compare/master...blam/templating-in-parameters)
+
+## Design Details
+
+There's a slight issue with the implementation of the `react-jsonschema-form`, which makes things like live updating on things like the `default` field slightly more difficult.
+Currently, on first render, the default value is populated and then stored in the `formData` object or the current state, and the default value is never re-evaluated again at a later stage.
+
+This means that if end users are wanting to set default values with `${{ parameters.myOtherProperty }}`, then they would need to ensure that they are on different steps in the form
+as the form would need to be re-rendered, and for performance reasons, we don't want to re-render the form on every `formData` update.
```
Comment (freben) [inline beps/0010-scaffolder-templating-in-parameters/README.md:163]: are debounced rerenders possible?

**Review by freben — APPROVED** [review]
Just some open thoughts, no need to block

File: beps/0010-scaffolder-templating-in-parameters/README.md (hunk 3 above)
Comment (benjdlambert) [reply beps/0010-scaffolder-templating-in-parameters/README.md:163]: Hmm, I'm not sure it helps with the overall problem though to be honest, I think it's just going to be a limitation that the default value is populated straight away on first render, not on submit. But maybe there's another way around this that we can work out later.

**Pull request merged**